src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/PseudoRoleToAccessProfileGenerator.java

```diff
@@ -19,7 +23,20 @@
 
     private static final String CASE_ROLE_ID_REGEX = "^(\\[[A-Za-z]+\\])$";
 
-    public List<RoleToAccessProfileDefinition> generate(CaseTypeDefinition caseTypeDefinition) {
+    private final CaseDefinitionRepository caseDefinitionRepository;
+
+    public PseudoRoleToAccessProfileGenerator(
+        @Qualifier(CachedCaseDefinitionRepository.QUALIFIER) CaseDefinitionRepository caseDefinitionRepository) {
+        this.caseDefinitionRepository = caseDefinitionRepository;
+    }
+
+    @Cacheable("caseTypePseudoRoleToAccessProfileCache")
+    public List<RoleToAccessProfileDefinition> generate(int version, String caseTypeId) {
+        CaseTypeDefinition caseTypeDefinition = caseDefinitionRepository.getCaseType(version, caseTypeId);
```

Comment:
instead of getting CaseTypeDefinition again, if we do something like this

@Cacheable(value = "caseTypePseudoRoleToAccessProfileCache", key = "#version")
List<RoleToAccessProfileDefinition> generate(int version, String caseTypeId, CaseTypeDefinition caseTypeDef) {